src/posit/connect/errors.py

```diff
@@ -1,15 +1,21 @@
+import json
+from typing import Any
+
+
 class ClientError(Exception):
     def __init__(
         self,
         error_code: int,
         error_message: str,
         http_status: int,
         http_message: str,
+        payload: Any = None,
     ):
-        self.error_code = error_code
-        self.error_message = error_message
-        self.http_status = http_status
-        self.http_message = http_message
-        super().__init__(
-            f"{error_message} (Error Code: {error_code}, HTTP Status: {http_status} {http_message})"
-        )
+        message = {
+            "error_code": error_code,
+            "error_message": error_message,
+            "http_status": http_status,
+            "http_message": http_message,
+            "payload": payload,
+        }
+        super().__init__(json.dumps(message))
```

Comment:
Since the payload is sometimes a JSON object, it seems appropriate to structure the entire message. My initial implementation appended the payload to the end of the existing string, but I thought that looked odd.

example, `error message (Error Code: 0, HTTP Status: 404 Foo Bar) { 'payload': 'object' }`